src/js/other-websites/inpage_menu.js

```diff
@@ -0,0 +1,278 @@
+    "popupRemainingAliases_2",
+    [numAliasesRemaining, maxNumAliases]
+  );
+
+  // Free user (who once was premium): Set text informing them how they have exceeded the maximum amount of aliases and cannot create any more
+  if (numAliasesRemaining < 0) {
+    remainingAliasesSpan.textContent = browser.i18n.getMessage(
+      "pageFillRelayAddressLimit"
+    );
+  }
+
+  // Premium user: Set text informing them how many aliases they have created so far
+  if (premium) {
+    remainingAliasesSpan.textContent = browser.i18n.getMessage(
+      "popupUnlimitedAliases",
+      [relayAddresses.length]
+    );
+  }
+
+  const maxNumAliasesReached = numAliasesRemaining <= 0;
+
+  // Create "Manage All Aliases" link
+  const relayMenuDashboardLink = document.querySelector(
+    ".fx-relay-menu-dashboard-link"
+  );
+  relayMenuDashboardLink.textContent = browser.i18n.getMessage("ManageAllAliases");
+  relayMenuDashboardLink.href = `${relaySiteOrigin}?utm_source=fx-relay-addon&utm_medium=input-menu&utm_content=manage-all-addresses`;
+  relayMenuDashboardLink.addEventListener("click", () => {
+    sendInPageEvent("click", "input-menu-manage-all-aliases-btn");
+    iframeCloseRelayInPageMenu();
+  });
+
+  // Create "Get unlimited aliases" link
+  getUnlimitedAliasesBtn.href = `${relaySiteOrigin}/premium?utm_source=fx-relay-addon&utm_medium=input-menu&utm_content=get-premium-link`;
+  getUnlimitedAliasesBtn.addEventListener("click", () => {
+    sendInPageEvent("click", "input-menu-get-premium-btn");
+    iframeCloseRelayInPageMenu();
+  });
+
+  if (!premium) {
+    if (maxNumAliasesReached) {
+      generateAliasBtn.remove();
+      sendInPageEvent("viewed-menu", "input-menu-max-aliases-message");
+      remainingAliasesSpan.textContent = browser.i18n.getMessage(
+        "pageFillRelayAddressLimit",
+        [numAliasesRemaining, maxNumAliases]
+      );
+      // Focus on "Get unlimited alias" button
+      getUnlimitedAliasesBtn.focus();
+    } else {
+      // Focus on "Generate New Alias" button
+      generateAliasBtn.focus();
+    }
+  } else {
+    // Focus on "Generate New Alias" button
+    generateAliasBtn.focus();
+    getUnlimitedAliasesBtn.remove();
+  }
+
+  //Check if premium features are available
+  const premiumCountryAvailability = (
+    await browser.storage.local.get("premiumCountries")
+  )?.premiumCountries;
+
+  if (
+    premiumCountryAvailability?.premium_available_in_country !== true ||
+    !maxNumAliasesReached
+  ) {
+    getUnlimitedAliasesBtn.remove();
+  }
+
+  // Handle "Generate New Alias" clicks
+  generateAliasBtn.addEventListener("click", async (generateClickEvt) => {
+    sendInPageEvent("click", "input-menu-generate-alias");
+    preventDefaultBehavior(generateClickEvt);
+
+    // Request the active tab from the background script and parse the `document.location.hostname`
+    const currentPageHostName = await browser.runtime.sendMessage({method: "getCurrentPageHostname"});
```

Comment:
praise: Ah, nice. So this is how to get the page's hostname while inside an iframe?